Sunrise/ViewModels/BaseViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright (c) 2016 Commercetools. All rights reserved.
+//
+
+import ReactiveCocoa
+import Result
+import ObjectMapper
+import Commercetools
+
+class BaseViewModel {
+
+    // Outputs
+    let alertMessageSignal: Signal<String, NoError>
+
+    let alertMessageObserver: Observer<String, NoError>
+
+    // MARK: Lifecycle
+
+    init() {
+        let (alertMessageSignal, alertMessageObserver) = Signal<String, NoError>.pipe()
+        self.alertMessageSignal = alertMessageSignal
+        self.alertMessageObserver = alertMessageObserver
+    }
+
+    func alertMessageForErrors(errors: [NSError]) -> String {
+        return errors.map({
+            var alertMessage = ""
+            if let failureReason = $0.userInfo[NSLocalizedFailureReasonErrorKey] as? String {
+                alertMessage += failureReason
+            }
+            if let localizedDescription = $0.userInfo[NSLocalizedDescriptionKey] as? String {
+                alertMessage += ": \(localizedDescription)"
+            }
```

Comment:
Somehow it's not working when trying to access `failureReason` and `localizedDescription` via the `userInfo`-dictionary directly. The string was always empty (see screenshot). 

![alert-1](https://cloud.githubusercontent.com/assets/1197899/15772049/23899448-296f-11e6-880f-b10c2808f1bd.png)



Changing it to the code below works finde.
```
            if let failureReason = $0.localizedFailureReason {
                alertMessage += failureReason
            }
            
            let localizedDescription = $0.localizedDescription
            alertMessage += ": \(localizedDescription)"
```

![alert-2](https://cloud.githubusercontent.com/assets/1197899/15772051/2785da0c-296f-11e6-9c3e-fe6f66166ef9.png)